drivers/sdhc/sam_hsmci.c

```diff
@@ -0,0 +1,538 @@
+/*
+ * Copyright 2023 Nikhef
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT atmel_sam_hsmci
+
+#include <zephyr/drivers/sdhc.h>
+#include <zephyr/drivers/gpio.h>
+#include <zephyr/devicetree.h>
+#include <zephyr/logging/log.h>
+#include <zephyr/kernel.h>
+#include <soc.h>
+#include <zephyr/drivers/pinctrl.h>
+#include <zephyr/drivers/clock_control.h>
+#include <zephyr/drivers/clock_control/atmel_sam_pmc.h>
+
+#ifndef HSMCI_MR_PDCMODE
+#error "Sorry, only SAM4E/SAM4S are supported at the moment"
```

Comment:
Are you sure that we can not drop this and make it compatible with remaining devices?
`hsmci->HSMCI_MR |= HSMCI_MR_PDCMODE;`

if not
```diff
+#error "The driver currently supports SAM4 SoC's"
```

Reply:
Please see my previous comment.